sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_endpoint_utils.py

```diff
@@ -218,7 +218,12 @@ def validate_scoring_script(deployment):
 
 
 def convert_v1_dataset_to_v2(output_data_set: DataVersion, file_name: str) -> Dict[str, Any]:
-    v2_dataset = UriFileJobOutput(
-        uri=f"azureml://datastores/{output_data_set.datastore_id}/paths/{output_data_set.path}/{file_name}"
-    ).serialize()
+    if file_name:
+        v2_dataset = UriFileJobOutput(
+            uri=f"azureml://datastores/{output_data_set.datastore_id}/paths/{output_data_set.path}/{file_name}"
+        ).serialize()
+    else:
+        v2_dataset = UriFileJobOutput(
```

Comment:
Is the dataset a folder in this case? Is there a `UriFolderJobOutput` class to use?

Reply:
Changing that gives me the following error:
` "message": "The Job request provided has the following validation errors. Please resolve them and try again.\nErrors:\n[\n  {\n    \"PropertyName\": \"OutputData.Values[0].JobOutputType\",\n    \"ErrorMessage\": \"Job output type must be 'uri_file'.\",\n    \"AttemptedValue\": \"uri_folder\",\n    \"ErrorCode\": \"PredicateValidator\"`